pkg/healthcheck/healthcheck.go

```diff
@@ -1854,6 +1869,43 @@ func (hc *HealthChecker) CheckNamespace(ctx context.Context, namespace string, s
 	return nil
 }
 
+func (hc *HealthChecker) checkClusterNetworks(ctx context.Context) error {
+	nodes, err := hc.kubeAPI.GetNodes(ctx)
+	if err != nil {
+		return err
+	}
+	var podCIDRs []string
+	for _, node := range nodes {
+		podCIDRs = append(podCIDRs, node.Spec.PodCIDR)
+	}
+	clusterNetworks := strings.Split(hc.linkerdConfig.ClusterNetworks, ",")
+	clusterIPNets := make([]*net.IPNet, len(clusterNetworks))
+	for i, clusterNetwork := range clusterNetworks {
+		_, ipNet, err := net.ParseCIDR(clusterNetwork)
+		if err != nil {
+			return err
+		}
+		clusterIPNets[i] = ipNet
+	}
+	for _, podCIDR := range podCIDRs {
+		podIP, _, err := net.ParseCIDR(podCIDR)
+		if err != nil {
+			return err
+		}
+		exists := false
+		for _, clusterIPNet := range clusterIPNets {
+			if clusterIPNet.Contains(podIP) {
+				exists = true
+				break
+			}
+		}
+		if !exists {
+			return fmt.Errorf("node podCIDR %s is not a default Kubernetes private network subnet.\nTry installing linkerd via linkerd install --set clusterNetworks=%s", podCIDR, podCIDR)
```

Comment:
I wouldn't speak of a "default Kubernetes private network subnet", because the cluster might (and probably has, and that could be the cause of the error) have been configured with a non-default subnet. If we accumulate all the errors, this message could recommend setting `clusterNetworks` to the list of all `podCIDR`s, but that would imply checking those aren't duped. I think it's enough to just say something like "set the clusterNetworks config appropriately during install`.

Reply:
How about something like the `node podCIDRs don't match the default linkerd configuration, to set the clusterNetworks config appropriately, install linkerd via linkerd --set clusterNetworks=<podCIDR>`?

Reply:
You don't know if the default linkerd config was used, so I think we should omit speaking of default configs either for kubernetes or linkerd.